contrib/ci/run-shellcheck.sh

```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/env bash
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+# Run ShellCheck on all bash scripts
+
+set -e
+set -x
+
+mapfile -t filestocheck < <(shfmt -f .)
+shellcheck -P SCRIPTDIR -x "${filestocheck[@]}"
```

Comment:
It will run shellcheck against all files found by `shfmt -f .`, which is run at the root of the project. It's pretty good at finding shell files, even if they don't have `.sh` filename extensions.

`SCRIPTDIR` is a special flag that causes shellcheck to use the directives `#shellcheck source=<path>` with a relative `<path>`. It's not a variable.